src/contracts/core/Slasher.sol

```diff
@@ -5,233 +5,475 @@ import "./SlasherStorage.sol";
-            } else if (slashingRequest.slashingRate == 0) {
-                // if the slashing rate is 0 due to cancellation, skip the slashing
-                continue;
-            }
-            
+
+            // calculate slashingRate based off allocated magnitude
+            uint64 slashingRate;
+            // uint64 slashingRate = 
+            //     uint64(bipsToSlash) * uint64(operatorSetManager.getSlashableBips(operator, operatorSet, strategy, epoch));
+            // calculate and apply new scaling factor
             uint64 scalingFactorBefore = shareScalingFactor(operator, strategy);
             uint64 scalingFactorAfter = SlashingAccountingUtils.findNewScalingFactor({
                 scalingFactorBefore: scalingFactorBefore,
-                rateToSlash: slashingRequest.slashingRate
+                rateToSlash: slashingRate
             });
             // update storage to reflect the slashing
-            slashedEpochHistory[operator][strategy].push(epoch);
+            slashingEpochHistory[operator][strategy].push(epoch);
+            slashingUpdates[operator][strategy][epoch] = SlashingUpdate({
+                slashingRate: slashingRate,
+                scalingFactor: scalingFactorAfter
+            });
             _shareScalingFactor[operator][strategy] = scalingFactorAfter;
             // TODO DONE?: note that this behavior risks off-by-one errors. it needs to be clearly defined precisely how the historical storage is supposed to work
-            slashingRequest.scalingFactor = scalingFactorAfter;
 
-            emit SlashingExecuted(epoch, operator, strategy, slashingRequest.slashingRate);
         }
     }
 
-    /// VIEW
+    /*******************************************************************************
+                                 INTERNAL FUNCTIONS
+    *******************************************************************************/
 
-    /**
-     * @notice fetches the requested parts per hundred million to slash for the
-     * given operator, strategy, epoch, and operator set
-     *
-     * @param operator the operator to get the requested slashing rate for
-     * @param strategy the strategy to get the requested slashing rate for
-     * @param operatorSet the operator set to get the requested requested slashing rate for
-     * @param epoch the epoch to get the requested slashing rate  for
-     *
-     * @return the requested parts per hundred million to slash for the given
-     * operator, strategy, epoch, and operator set
-     *
-     * @dev may exceed the AVS operator set's allowed slashing per epoch;
-     * the `getPendingSlashedPPHM` will accurately reflect this ceiling though.
-     */
-    function getRequestedSlashingRate(
+    /// @notice push or update the latest magnitude update for the given operator, strategy, and operatorSet
+    /// depending on the current effectTimestamp
+    function _updateMagnitudeHistory(
         address operator,
```

Comment:
you can pass storage mappings as parameters to internal functions. that would probably be cleaner here